### DISCUSSION TITLE
how iphone recognizes duplicates - iphone sync + sync of the same objects with cli

### DISCUSSION THREAD

**Discussion by eleaner** [immich-app/immich#13706] [Q&A]
Hi,

I have quite a clutter of photos and videos on my phone and icloud
i also have offline backup and I would like to move to immich (which is awesome so far)

so I uploaded the whole offline library via cli and ind of expected iPhone to recognize it.
instead it painfully goes one by one through 10k of objects
is it normal? 
is it trying to upload them to server anyway in hope that server will recognize duplicates?
or is it just the process of downloading them from the cloud and iPhone app recognizes duplicates without uploading?

**Comment by bo0tzz** [comment]
>  is it trying to upload them to server anyway in hope that server will recognize duplicates?

Yes, that is currently how it works. It'll be improved in the future.
